generate_tidal_profile.py

```diff
@@ -0,0 +1,438 @@
+        max_profile_num = np.where(total_energy == max(total_energy))[0][0]
+        min_profile_num = np.where(total_energy == min(total_energy))[0][0]
+
+        # Plot the profiles with min and max energy
+        fig = plt.figure()
+        ax1 = fig.add_subplot(121)
+        self.power_profiles[max_profile_num].plot(
+            ax=ax1, title='Profile with max energy generation')
+        ax1.set_ylabel('Power (kW)')
+        ax2 = fig.add_subplot(122)
+        self.power_profiles[min_profile_num].plot(
+            ax=ax2, title='Profile with min energy generation')
+        ax2.set_ylabel('Power (kW)')
+
+    def get_dc_to_ac(self):
+        """ Returns the dc to ac ratio. """
+
+        # Should we keep this? I don't think this matters
+        # for tidal as much for solar
+
+        # Get DC power output
+        dc = self.advanced_inputs['rated_power']
+
+        # Get inverter AC power output
+        ac = self.advanced_inputs['inverter_output']
+
+        return dc / ac
+
+    def get_turbine_losses(self):
+        """ Returns the system losses. """
+
+        # I don't think this is necessary either, since losses are
+        # an input, not an output like for solar
+
+        turbine_losses = self.advanced_inputs['turbine_losses']
+
+        return turbine_losses
+
+
+def calc_tidal_prod(tidal_profile, latitude, longitude,
+                    rated_power, rotor_length, turbine_number,
+                    inverter_efficiency, maximum_cp, turbine_losses,
+                    cut_in_velocity, cut_out_velocity, validate=False):
+
+    """ Calculates the production from a tidal profile. """
+
+    if validate:
+        # Put arguments in a dict
+        args_dict = {'tidal_profile': tidal_profile,
+                     'latitude': latitude,
+                     'longitude': longitude,
+                     'rated_power': rated_power,
+                     'rotor_length': rotor_length,
+                     'turbine_number': turbine_number,
+                     'inverter_efficiency': inverter_efficiency,
+                     'maximum_cp': maximum_cp,
+                     'turbine_losses': turbine_losses,
+                     'cut_in_velocity': 0.7,
+                     'cut_out_velocity': 2.6}
+    #
+        # Validate all parameters
+        validate_all_parameters(args_dict)
+
+
+    # Calculate DC power (normalized to turbine size. i.e. per 1kW of tidal)
+
+    dc_power = pd.DataFrame()
+
+    for index, row in tidal_profile.iterrows():
+        u = row['v']
+        if u > cut_in_velocity and u < cut_out_velocity:
+            dc_power.at[
+                index, 'power'] = 0.5 * maximum_cp * u ** 3 * np.pi * rotor_length ** 2 / (rated_power * turbine_number)
+        elif u < cut_in_velocity:
+            dc_power.at[index, 'power'] = 0
+        elif u > cut_out_velocity:
+            dc_power.at[index, 'power'] = rated_power / (rated_power * turbine_number)
```

Comment:
Is this equation correct?

Reply:
If tidal turbines operate similarly to wind turbines, than my assumption is that the the capacity of a specific turbine (e.g. 50kW) is equal to the peak production of that turbine. So if we we're above the cut-out speed, the turbine is limited to operating at max production. Assuming a 50kW turbine, if above cut-out speed, the power equation outputs that the turbine is producing a continuous 50kW of power. However, I am trying to calculate power output per 1kW of tidal, which is why the output is divided by the rated power of the turbine times how many turbines there are